nectar/downloaders/threaded.py

```diff
@@ -215,10 +215,14 @@ def _fetch(self, request, session):
                             url=request.url)
                         )
                         _logger.warning(msg)
-                    response = session.get(
-                        request.url, headers=headers, timeout=(self.config.connect_timeout,
-                                                               self.config.read_timeout)
-                    )
+
+                    response = session.get(request.url, headers=headers,
+                                           timeout=(self.config.connect_timeout,
+                                                    self.config.read_timeout))
+
+                    report.headers = response.headers
```

Comment:
I have also confirmed this functionality. The `response = session.get(...)` does not return until the headers are already received by `python-requests`. My concerns are resolved.